docs/reference/ImageMorph.rst

```diff
@@ -4,7 +4,149 @@
+same size, and Pillow builds a lookup table from them using
+:class:`LutBuilder`.
+
+
+Using :class:`LutBuilder`
+-------------------------
+
+The :class:`LutBuilder` class constructs a LUT that defines how a morphological
+operation should behave. A LUT maps every possible 3×3 neighborhood around a
+pixel to an output pixel value (either “on” or “off”).
+
+Basic uses like dilation and erosion can be created by specifying
+preset names (``"dilation4"``, ``"dilation8"``, ``"erosion4"``,
+``"erosion8"``, ``"edge"``), or you may define custom patterns.
+
+Example: creating a LUT for a 2×2 corner detector::
+
+    from PIL import ImageMorph
+
+    patterns = [
+        "10",
+        "11",
+    ]
+
+    lb = ImageMorph.LutBuilder(op_name="corner")
+    lb.add_patterns(patterns)
+    lut = lb.build_lut()
+
+You can inspect, save, or reuse the LUT with :meth:`LutBuilder.get_lut`,
+:meth:`LutBuilder.load_lut`, or :meth:`LutBuilder.save_lut`.
+
+
+Applying morphology with :class:`MorphOp`
+-----------------------------------------
+
+Once a LUT is created, the :class:`MorphOp` class applies it to an image. The
+:meth:`MorphOp.apply` method performs the morphological operation and returns
+a tuple ``(count, out_image)`` where:
+
+* ``count`` is the number of pixels that changed, and
+* ``out_image`` is the resulting processed image.
+
+Example: applying a simple dilation operation::
+
+    from PIL import Image, ImageMorph
+
+    im = Image.open("input.png").convert("L")
+
+    # Built-in 8-connected dilation
+    op = ImageMorph.MorphOp(op_name="dilation8")
+
+    count, out = op.apply(im)
+    out.save("dilated.png")
+
+You could also use the method :meth:`MorphOp.match` to check where a pattern
+matches without modifying the image, and :meth:`MorphOp.get_on_pixels` to
+get the coordinates of “on” pixels after pattern matching.
+
+Example: pattern matching without modifying the image::
+
+    op = ImageMorph.MorphOp(op_name="edge")
+    result = op.match(im)
+
+    # result is a list of (x, y) coordinates
+    print("Edge pixels found:", len(result))
+
+
+Saving and loading LUTs
+-----------------------
+
+LUTs created by :class:`LutBuilder` can be serialized and reused later. This
+is helpful when repeatedly applying the same pattern in a batch-processing
+workflow.
+
+Example::
+
+    lb = ImageMorph.LutBuilder(op_name="custom")
+    lb.add_patterns(patterns)
+    lb.build_lut()
+    lb.save_lut("custom.lut")
+
+    # Later...
+    op = ImageMorph.MorphOp()
+    op.load_lut("custom.lut")
+    count, out = op.apply(im)
+
+
+Relationship to `_imagingmorph`
+-------------------------------
+
+The :mod:`PIL.ImageMorph` module provides a Python interface, while the
+low-level implementation of morphological lookup table evaluation is handled
+by the C extension module :mod:`_imagingmorph`. Users should interact
+with :class:`LutBuilder` and :class:`MorphOp`, not with the C extension
+directly.
```

Comment:
I'm interested, why did you feel the need to mention `_imagingmorph`? `_imagingft` and `_imagingmath` aren't mentioned in our docs, for precisely the reason you mention here - we don't want users to deal with them directly. How do you think users would discover `_imagingmorph` in the first place?